addons/hr_expense/wizard/account_payment_register.py

```diff
@@ -9,6 +9,15 @@ class AccountPaymentRegister(models.TransientModel):
     # -------------------------------------------------------------------------
     # BUSINESS METHODS
     # -------------------------------------------------------------------------
+    
+    @api.model
+    def _get_line_batch_key(self, line):
+        # OVERRIDE to set the bank account defined on the employee
+        res = super()._get_line_batch_key(line)
+        expense_sheet = self.env['hr.expense.sheet'].search([('account_move_id', 'in', line.move_id.ids)])
```

Comment:
Something is missing in tthe domain: the expense sheet can be paid by the employee or by the company. We only want to use the employee's bank account when it's paid by the employee, so I would add something in this domain to only consider the ones whose related expense has payment_mode = 'company_account'

Reply:
Done, thanks !

I guess you mean 'own_account' instead of 'company_account'.
Also, in case of payment_mode = 'company_account', the expense goes directly from Approved to Paid, so we should normally not end up in the payment.register. But the check still worths the case I think.

Reply:
Indeed, own_account ; sorry ^^